File: codingTest/baekjoon/javascript/1874.js
var inp = require("fs").readFileSync("/dev/stdin").toString().split("\n");

var N = +inp.shift();
var stack = [];
var result = "";
var count = 0;
for (let i = 1; i <= N; i++) {
  stack.push(i);
  result += "+\n";
  while (count < N && stack[stack.length - 1] == +inp[count]) {
    stack.pop();
    count++;
    result += "-\n";
  }
}
if (stack.length) console.log("NO");
else console.log(result);

File: codingTest/baekjoon/python/1874.py
from sys import stdin

N = int(stdin.readline())
arr = []
for _ in range(N):
    arr.append(int(stdin.readline()))
arr.reverse()

stack = []
result = []
for i in range(1, N + 1):
    stack.append(i)
    result.append("+")
    while len(stack) != 0 and stack[-1] == arr[-1]:
        stack.pop()
        arr.pop()
        result.append("-")
if len(arr) != 0:
    print("NO")
else:
    print("\n".join(result))
